packages/backend/server/src/plugins/oauth/controller.ts

```diff
@@ -43,14 +44,15 @@ export class OAuthController {
   @Post('/preflight')
   @HttpCode(HttpStatus.OK)
   async preflight(
-    @Body('provider') unknownProviderName?: string,
-    @Body('redirect_uri') redirectUri?: string
+    @Body('provider') unknownProviderName?: keyof typeof OAuthProviderName,
+    @Body('redirect_uri') redirectUri?: string,
+    @Body('client') client?: string,
+    @Body('client_nonce') clientNonce?: string
```

Comment:
magiclink requires the same protection

Reply:
https://app.graphite.dev/github/pr/toeverything/AFFiNE/10736/refactor(server)-improve-magic-link-login-flow